### DISCUSSION TITLE
how to pass new_symblic_buffer to wasm?

### DISCUSSION THREAD

**Discussion by dengliangjun** [trailofbits/manticore#2448] [Q&A]
I found new_symblic_buffer is defined in wasm/state.py
so for as I known, if wasm function parameter is  a string, parameter is donated by "offset and length". 
I'm new at Manticore, how to use new_symblic_buffer  to pass parameter?
what's more, where is the offset defined?
Thanks for your reading, can anyone help me?

**Comment by jdw1023** [comment]
Hi, did you figure out how to pass string/new_symblic_buffer ?

**Comment by dengliangjun** [comment]
u need to copy string to webassembly memory, that is linear memory space, pass the start position as interger to webassembly, then read data according to position and length





来自skycto.com企业邮箱------------------------------------------------------------------
***@***.***>
日　期：2021年12月10日 13:54:17
***@***.***>
***@***.***>; ***@***.***>
主　题：Re: [trailofbits/manticore] how to pass new_symblic_buffer to wasm? (Discussion #2448)

Hi, did you figure out how to pass string/new_symblic_buffer ?
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.
